.github/workflows/vast.yaml

```diff
@@ -140,110 +142,107 @@ jobs:
+          run_if_changed changelog "CHANGELOG.md" "changelog/" "web/" "python/"
+          VAST_SOURCES=(cmake/ CMakeLists.txt libvast/ libvast_test/ scripts/ schema/ tools/ vast/ vast.yaml.example version.json)
+          run_if_changed vast "$VAST_SOURCES[@]"
+          run_if_changed vast-plugins "plugins/" "contrib/vast-plugins/"
+          # The vast-plugins job downloads the VAST artifact from GCS, so we
+          # need to run the vast job in case it is missing.
           if [[ ${run_vast_plugins} == "true" && ${run_vast} == "false" ]]; then
             deb_package="$(echo "vast-${before_version}-linux-Release-GCC" | awk '{ print tolower($0) }')"
             mac_package="$(echo "vast-${before_version}-darwin-Release-Clang" | awk '{ print tolower($0) }')"
-            if ! gsutil -q stat gs://${{ secrets.GCS_BUCKET }}/${deb_package}.tar.gz; then
-              run_vast=true
-            elif ! gsutil -q stat gs://${{ secrets.GCS_BUCKET }}/${mac_package}.tar.gz; then
+            if ! gsutil -q stat gs://${{ secrets.GCS_BUCKET }}/${deb_package}.tar.gz || \
+               ! gsutil -q stat gs://${{ secrets.GCS_BUCKET }}/${mac_package}.tar.gz; then
+              echo "run-vast=true" >> $GITHUB_OUTPUT
               run_vast=true
             fi
           fi
-          echo "run-vast=${run_vast}" >> $GITHUB_OUTPUT
-          run_docker_compose=${run_vast}
-          if [[ $GITHUB_EVENT_NAME == "workflow_dispatch" ]]; then
-            run_docker_compose=false
-          elif [[ ${run_vast_plugins} == "true" || ${run_python} == "true" ]]; then
-            run_docker_compose=true
-          elif ! is_unchanged docker/ Dockerfile .dockerignore; then
-            run_docker_compose=true
-          fi
-          echo "run-docker-compose=${run_docker_compose}" >> $GITHUB_OUTPUT
-          run_docker_vast=${run_docker_compose}
-          if [[ $GITHUB_EVENT_NAME == "workflow_dispatch" ]]; then
-            run_docker_vast=false
-          elif [[ ${run_vast_io} == "true" || ${run_python} == "true" ]]; then
-            run_docker_vast=true
-          fi
-          echo "run-docker-vast=${run_docker_vast}" >> $GITHUB_OUTPUT
-          run_vast_nix=${run_vast}
-          if ${run_vast_plugins}; then
-            run_vast_nix=true
-          elif ! is_unchanged nix/ flake.lock *.nix; then
-            run_vast_nix=true
-          elif [[ $GITHUB_EVENT_NAME == "workflow_dispatch" ]]; then
-            run_vast_nix=true
-          fi
-          echo "run-vast-nix=${run_vast_nix}" >> $GITHUB_OUTPUT
+          run_docker_compose="$(any ${run_vast} ${run_vast_plugins} ${run_python})"
+          run_if_changed_default docker-compose $run_docker_compose "docker/" "Dockerfile" ".dockerignore"
+          run_docker_vast="$(any ${run_docker_compose} ${run_vast_io} ${run_python})"
+          run_if_changed_default docker-vast $run_docker_vast
+          run_if_changed_default regression-tests $run_docker_vast
```

Comment:
I realized these invocations with out passing paths aren't tested yet.
Just leaving that as a comment for myself.